backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -179,6 +195,23 @@ def update_collection(collection_uuid: str, body: dict, user: str):
     )
     if not collection:
         raise ForbiddenHTTPException()
+
+    # Compute the diff between old and new DOI
+    old_doi = collection.get_normalized_doi()
+    new_doi = _get_doi_from_body(body)
+
+    if old_doi and not new_doi: 
+        # If the DOI was deleted, remove the publisher_metadata field
+        collection.update(publisher_metadata = None)
+    elif new_doi != old_doi:
+        # If the DOI has changed, fetch and update the metadata
+        provider = crossref_provider.CrossrefProvider()
+        try:
+            publisher_metadata = provider.fetch_metadata(new_doi)
+        except Exception:
+            publisher_metadata = None
```

Comment:
I generally like to defer the exception handling to the caller, as it gives you more control. I agree that, for this specific case, handling the exception inside the method would be easier, but that would also make the provider less reusable.

Reply:
agreed re:caller handling, but I also like to have the exception-throwing method raise application specific exceptions, rather than specific 3rd party library exception. but pretty minor concern in this simple case.

Reply:
Yes, that's something I can add. I will also add a warning.